src/stream_factory.js

```diff
@@ -23,7 +23,7 @@ class StreamFactory {
 
     this.controlStream = this._pipe(new ControlStream(ChunkStreamParams))
     this.userControlStream = this._pipe(new UserControlStream())
-    this.netConnection = this._pipe(new NetConnection())
+    this.netConnection = this._pipe(new NetConnection(socket))
```

Comment:
This approach goes against Stream Pipeline pattern.

The socket is the lowest layer of the pipeline and `NetConnection` must be independent of it (I wanted to separate concerns here), so constructor must not receive any parameter referencing lower layers because it is like 'short-circuiting' the pipeline. In fact, none of the entities, except Stream factory, is aware of the socket.

I am not worried about closing a connection, but in case of need, the approach should make use of `.end()` or `.destroy()` methods on Duplex Streams, listen to the triggered events and do whatever is needed on the rest of the piped entities.